app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,49 @@
+class AnswersController < ApplicationController
+
+  def new
+    @question = Question.find(params[:id])
+    @answer = @question.answers.new
+  end
+
+  def create
+    @question = Question.find(params[:question_id])
+    @answer = @question.answers.build(answer_params)
+    @answer.user_id = current_user.id
+    if @answer.save
+      redirect_to "/questions/#{@answer.question.id}"
+    else
+      flash[:alert] = @answer.errors.full_messages
+      render 'new'
+  end
+end
+
+  def edit
+    @answer = Answer.find(params[:id])
+    @question = Question.find(params[:question_id])
```

Comment:
@question is already there as @answer.question - you don't need the second query.